From: Rolf Eike Beer <eike@sf-mail.de>
Date: Fri, 13 Apr 2012 08:50:49 +0200
Subject: FindPkgConfig.cmake: fix documented output variable not set
Origin: backport, commit:3ea850a5023060b84dcc0e6f0098c32c28b15807
Bug: http://public.kitware.com/Bug/view.php?id=13125
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=669627
Applied-Upstream: 2.8.9

The real fix is from Yury G. Kudryashov while I added the surrounding cleanups.
An additional hint to really get this fixed came from Rex Dieter.

--- a/Modules/FindPkgConfig.cmake
+++ b/Modules/FindPkgConfig.cmake
@@ -13,11 +13,10 @@
 # When the 'QUIET' argument is set, no status messages will be printed.
 #
 # It sets the following variables:
-#   PKG_CONFIG_FOUND          ... true if pkg-config works on the system
+#   PKG_CONFIG_FOUND          ... if pkg-config executable was found
 #   PKG_CONFIG_EXECUTABLE     ... pathname of the pkg-config program
 #   PKG_CONFIG_VERSION_STRING ... the version of the pkg-config program found
 #                                 (since CMake 2.8.8)
-#   PKG_CONFIG_FOUND          ... if pkg-config executable was found
 #
 # For the following variables two sets of values exist; first one is the
 # common one and has the given PREFIX. The second set contains flags
@@ -104,6 +103,11 @@ find_package_handle_standard_args(PkgCon
                                   REQUIRED_VARS PKG_CONFIG_EXECUTABLE
                                   VERSION_VAR PKG_CONFIG_VERSION_STRING)
 
+# This is needed because the module name is "PkgConfig" but the name of
+# this variable has always been PKG_CONFIG_FOUND so this isn't automatically
+# handled by FPHSA.
+set(PKG_CONFIG_FOUND "${PKGCONFIG_FOUND}")
+
 # Unsets the given variables
 macro(_pkgconfig_unset var)
   set(${var} "" CACHE INTERNAL "")
--- a/Tests/CMakeOnly/AllFindModules/CMakeLists.txt
+++ b/Tests/CMakeOnly/AllFindModules/CMakeLists.txt
@@ -71,7 +71,7 @@ endmacro(check_version_string)
 # reported.
 
 foreach(VTEST ALSA ARMADILLO BZIP2 CUPS CURL EXPAT FREETYPE GETTEXT GIT HSPELL
-        JASPER LIBXML2 LIBXSLT PERL PostgreSQL TIFF ZLIB)
+        JASPER LIBXML2 LIBXSLT PERL PKG_CONFIG PostgreSQL TIFF ZLIB)
     check_version_string(${VTEST} ${VTEST}_VERSION_STRING)
 endforeach(VTEST)
 
@@ -82,4 +82,3 @@ endforeach(VTEST)
 
 check_version_string(PYTHONINTERP PYTHON_VERSION_STRING)
 check_version_string(SUBVERSION Subversion_VERSION_SVN)
-check_version_string(PKGCONFIG PKG_CONFIG_VERSION_STRING)
